File: LLM-Translations/GPT-4/codenet/Java/Go/s580342216.go

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	for scanner.Scan() {
		line := scanner.Text()
		runCase(line)
	}
}

func runCase(line string) {
	split := strings.Fields(line)
	N, _ := strconv.Atoi(split[0])
	K, _ := strconv.Atoi(split[1])

	len := 0
	for N > 0 {
		N /= K
		len++
	}

	fmt.Println(len)
}

func strToIntArray(str string) []int {
	vals := strings.Fields(str)
	sz := len(vals)
	res := make([]int, sz)
	for i := 0; i < sz; i++ {
		res[i], _ = strconv.Atoi(vals[i])
	}
	return res
}

type LCS struct {
	dp [][]int
}

func (l *LCS) lcs(A, B string) int {
	szA := len(A)
	szB := len(B)
	l.dp = make([][]int, szA+1)
	for i := range l.dp {
		l.dp[i] = make([]int, szB+1)
	}

	for i := 0; i <= szA; i++ {
		l.dp[i][0] = 0
	}
	for j := 0; j <= szB; j++ {
		l.dp[0][j] = 0
	}

	for i := 1; i <= szA; i++ {
		for j := 1; j <= szB; j++ {
			if A[i-1] == B[j-1] {
				l.dp[i][j] = l.dp[i-1][j-1] + 1
			} else {
				l.dp[i][j] = max(l.dp[i-1][j], l.dp[i][j-1])
			}
		}
	}

	return l.dp[szA][szB]
}

func (l *LCS) getLCS(X, Y string) string {
	m := len(X)
	n := len(Y)

	index := l.dp[m][n]
	temp := index

	lcs := make([]byte, index+1)
	lcs[index] = 0

	i := m
	j := n
	for i > 0 && j > 0 {
		if X[i-1] == Y[j-1] {
			lcs[index-1] = X[i-1]
			i--
			j--
			index--
		} else if l.dp[i-1][j] > l.dp[i][j-1] {
			i--
		} else {
			j--
		}
	}

	return string(lcs[:temp+1])
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

// 

File: LLM-Translations/GPT-4/codenet/Java/C/s580342216.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void run_case();
int* strToIntArray(char *str, int sz);
int lcs(char *A, char *B, int sz_a, int sz_b, int** dp);
char* getLCS(char *X, char *Y, int m, int n, int** dp);

int main() {
    char line[1024];

    while (fgets(line, sizeof(line), stdin)) {
        if(line[0]=='\n') continue;
        run_case(line);
    }

    return 0;
}

void run_case(char *line) {
    int N, K;

    sscanf(line, "%d %d", &N, &K);

    int len = 0;
    while (N > 0) {
        N /= K;
        len++;
    }

    printf("%d\n", len);
    return;
}

int* strToIntArray(char *str, int sz) {
    int *res = (int *) malloc(sz * sizeof(int));
    int i = 0;
    char *token = strtok(str, " ");
    while (token != NULL) {
        res[i++] = atoi(token);
        token = strtok(NULL, " ");
    }
    return res;
}

int lcs(char *A, char *B, int sz_a, int sz_b, int** dp) {
    for (int i = 0; i <= sz_a; i++) dp[i][0] = 0;
    for (int j = 0; j <= sz_b; j++) dp[0][j] = 0;

    for (int i = 1; i <= sz_a; i++) {
        for (int j = 1; j <= sz_b; j++) {
            if (A[i - 1] == B[j - 1]) {
                dp[i][j] = dp[i - 1][j - 1] + 1;
            } else {
                dp[i][j] = (dp[i - 1][j] > dp[i][j - 1]) ? dp[i - 1][j] : dp[i][j - 1];
            }
        }
    }

    return dp[sz_a][sz_b];
}

char* getLCS(char *X, char *Y, int m, int n, int** dp) {
    int index = dp[m][n];
    int temp = index;

    char *lcs = (char *) malloc((index + 1) * sizeof(char));
    lcs[index] = '\0';

    int i = m;
    int j = n;
    while (i > 0 && j > 0) {
        if (X[i - 1] == Y[j - 1]) {
            lcs[index - 1] = X[i - 1];
            i--;
            j--;
            index--;
        } else if (dp[i - 1][j] > dp[i][j - 1]) {
            i--;
        } else {
            j--;
        }
    }

    return lcs;
}

// 
